server/forge/gitlab/gitlab.go

```diff
@@ -553,28 +553,26 @@ func (g *GitLab) Deactivate(ctx context.Context, user *model.User, repo *model.R
 		}
 
 		for _, hook := range hooks {
-			if hook.URL == webURL {
+			if strings.Contains(hook.URL, webURL) {
 				hookID = hook.ID
-				break
+				_, err = client.Projects.DeleteProjectHook(_repo.ID, hookID, gitlab.WithContext(ctx))
+				log.Info().Msg(fmt.Sprintf("successfully deleted hook with ID %d for repo %s", hookID, repo.FullName))
+				if err != nil {
+					return err
+				}
 			}
 		}
 
 		// Exit the loop when we've seen all pages
-		if resp.CurrentPage >= resp.TotalPages {
+		if hookID != -1 || resp.CurrentPage >= resp.TotalPages {
```

Comment:
Looks like it would still stop after the first page with a match `hookID != -1`? 😉 

```suggestion
		if resp.CurrentPage >= resp.TotalPages {
```

Reply:
OK yes, in this new format we can just skim through all pages and don't need a custom exit logic.